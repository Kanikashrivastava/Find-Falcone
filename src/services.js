const BACKEND_URL = process.env.REACT_APP_BACKEND_URL;

const makeCall = (url, option) => {
    console.log(`${BACKEND_URL}${url}`)
    return fetch(`${BACKEND_URL}${url}`, option).then((response) => response.json())
}
const httpService = {
    get: (url, options = {}) => makeCall(url, {
        ...options,
        method: 'GET',
        headers: {
            ...options.headers,
            'Content-Type': 'application/json;charset=utf-8'
        },

    }),
    post: (url, options = {}) => makeCall(url, {
        ...options,
        method: 'POST',
        body: options.payload,
        headers: {
            ...options.headers,
            'Accept': 'application/json',
            'Content-Type': 'application/json;charset=utf-8'
        },

    })
}
export const fetchToken = () => {
    return httpService.post('/token')
}
export const findFalacon = () => {

}
export const fetchVehicles = () => {

}
export const fetchPlanets = () => {

}