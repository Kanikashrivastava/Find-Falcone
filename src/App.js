import React, { useEffect, useState } from 'react';
import { DestinationVehicleBook } from './components/Planet'
import { fetchToken } from './services';
const useToken = () => {
  const [token, setToken] = useState('');
  useEffect(() => {
    fetchToken().then((resp) => setToken(resp.token))
  }, [])
  return token;
}

function App() {
  const token = useToken();
  const planets = [
    { name: 'Planet 1' }, 
    { name: 'Planet 2' }, 
    { name: 'Planet 3' }, 
    { name: 'Planet 4' }
  ];
  const vehicles = [
    { name: 'Vehicles 1', total_no: 1 }, 
    { name: 'Vehicles 2', total_no: 2 }, 
    { name: 'Vehicles 3', total_no: 1 }, 
    { name: 'Vehicles 4', total_no: 1 }
  ];
  return (
    <div className="App">
      <header className="App-header">
        Find Falcone
      </header>
      <DestinationVehicleBook optionNum="1" planets={planets} vehicles={vehicles} />
      <DestinationVehicleBook optionNum="2" planets={planets} vehicles={vehicles} />
      <DestinationVehicleBook optionNum="3" planets={planets} vehicles={vehicles} />
      <DestinationVehicleBook optionNum="4" planets={planets} vehicles={vehicles} />
    </div>
  );
}

export default App;
