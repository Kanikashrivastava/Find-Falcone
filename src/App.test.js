import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import App from './App';
import { fetchPlanets } from './services'

jest.mock('./services.js')

fetchPlanets.mockResolvedValue([
  
])

test('renders learn react link', () => {
  const { asFragment, getAllByTestId} = render(<App />);
  fireEvent.click(getAllByTestId("planet-select")[0])
  fireEvent.click(getAllByTestId("planet-select")[1])
  fireEvent.click(getAllByTestId("planet-select")[2])
  fireEvent.click(getAllByTestId("planet-select")[3])
  expect(asFragment()).toMatchSnapshot()
});

