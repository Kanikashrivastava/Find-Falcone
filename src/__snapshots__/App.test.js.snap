// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      Find Falcone
    </header>
    <div
      class="MuiFormControl-root makeStyles-formControl-1"
    >
      <label
        id="demo-simple-select-label"
      >
        Destination 1
      </label>
      <select
        class="makeStyles-select-3"
        data-testid="planet-select"
        id="planet-select"
      >
        <option
          value="default"
        >
          Select Planet
        </option>
        <option
          value="Planet 1"
        >
          Planet 1
        </option>
        <option
          value="Planet 2"
        >
          Planet 2
        </option>
        <option
          value="Planet 3"
        >
          Planet 3
        </option>
        <option
          value="Planet 4"
        >
          Planet 4
        </option>
      </select>
    </div>
    <div
      class="MuiFormControl-root makeStyles-formControl-1"
    >
      <label
        id="demo-simple-select-label"
      >
        Destination 2
      </label>
      <select
        class="makeStyles-select-3"
        data-testid="planet-select"
        id="planet-select"
      >
        <option
          value="default"
        >
          Select Planet
        </option>
        <option
          value="Planet 1"
        >
          Planet 1
        </option>
        <option
          value="Planet 2"
        >
          Planet 2
        </option>
        <option
          value="Planet 3"
        >
          Planet 3
        </option>
        <option
          value="Planet 4"
        >
          Planet 4
        </option>
      </select>
    </div>
    <div
      class="MuiFormControl-root makeStyles-formControl-1"
    >
      <label
        id="demo-simple-select-label"
      >
        Destination 3
      </label>
      <select
        class="makeStyles-select-3"
        data-testid="planet-select"
        id="planet-select"
      >
        <option
          value="default"
        >
          Select Planet
        </option>
        <option
          value="Planet 1"
        >
          Planet 1
        </option>
        <option
          value="Planet 2"
        >
          Planet 2
        </option>
        <option
          value="Planet 3"
        >
          Planet 3
        </option>
        <option
          value="Planet 4"
        >
          Planet 4
        </option>
      </select>
    </div>
    <div
      class="MuiFormControl-root makeStyles-formControl-1"
    >
      <label
        id="demo-simple-select-label"
      >
        Destination 4
      </label>
      <select
        class="makeStyles-select-3"
        data-testid="planet-select"
        id="planet-select"
      >
        <option
          value="default"
        >
          Select Planet
        </option>
        <option
          value="Planet 1"
        >
          Planet 1
        </option>
        <option
          value="Planet 2"
        >
          Planet 2
        </option>
        <option
          value="Planet 3"
        >
          Planet 3
        </option>
        <option
          value="Planet 4"
        >
          Planet 4
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;
